nifi-external/nifi-kafka-connect/nifi-kafka-connector/src/main/java/org/apache/nifi/kafka/connect/StatelessKafkaConnectorUtil.java

```diff
@@ -0,0 +1,332 @@
+        final String truststoreFile = properties.get(TRUSTSTORE_FILE);
+        if (truststoreFile == null || truststoreFile.trim().isEmpty()) {
+            return null;
+        }
+
+        final SslContextDefinition sslContextDefinition;
+        sslContextDefinition = new SslContextDefinition();
+        sslContextDefinition.setTruststoreFile(truststoreFile);
+        sslContextDefinition.setTruststorePass(properties.get(TRUSTSTORE_PASSWORD));
+        sslContextDefinition.setTruststoreType(properties.get(TRUSTSTORE_TYPE));
+
+        final String keystoreFile = properties.get(KEYSTORE_FILE);
+        if (keystoreFile != null && !keystoreFile.trim().isEmpty()) {
+            sslContextDefinition.setKeystoreFile(keystoreFile);
+            sslContextDefinition.setKeystoreType(properties.get(KEYSTORE_TYPE));
+
+            final String keystorePass = properties.get(KEYSTORE_PASSWORD);
+            sslContextDefinition.setKeystorePass(keystorePass);
+
+            final String explicitKeyPass = properties.get(KEY_PASSWORD);
+            final String keyPass = (explicitKeyPass == null || explicitKeyPass.trim().isEmpty()) ? keystorePass : explicitKeyPass;
+            sslContextDefinition.setKeyPass(keyPass);
+        }
+
+        return sslContextDefinition;
+    }
+
+    private static URLClassLoader getConnectClassLoader() {
+        final ClassLoader classLoader = StatelessKafkaConnectorUtil.class.getClassLoader();
+        if (!(classLoader instanceof URLClassLoader)) {
+            throw new IllegalStateException("No configuration value was set for the " + NAR_DIRECTORY + " configuration property, and was unable to determine the NAR directory automatically");
+        }
+
+        return (URLClassLoader) classLoader;
+    }
+
+    private static File detectBootstrapJar() {
+        final URLClassLoader urlClassLoader = getConnectClassLoader();
+        for (final URL url : urlClassLoader.getURLs()) {
+            final String artifactFilename = url.getFile();
+            if (artifactFilename == null) {
+                continue;
+            }
+
+            final File artifactFile = new File(artifactFilename);
+            if (STATELESS_BOOTSTRAP_FILE_PATTERN.matcher(artifactFile.getName()).matches()) {
+                return artifactFile;
+            }
+        }
+
+        return null;
+    }
+
+    private static File detectNarDirectory() {
+        final File bootstrapJar = detectBootstrapJar();
+        if (bootstrapJar == null) {
+            final URLClassLoader urlClassLoader = getConnectClassLoader();
+            logger.error("ClassLoader that loaded Stateless Kafka Connector did not contain nifi-stateless-bootstrap. URLs that were present: {}", Arrays.asList(urlClassLoader.getURLs()));
+            throw new IllegalStateException("No configuration value was set for the " + NAR_DIRECTORY + " configuration property, and was unable to determine the NAR directory automatically");
```

Comment:
This message appears to be the same as the one used in `getConnectClassLoader`.  The message could be declared once, or this message could be adjusted.